webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js

```diff
@@ -5,22 +5,26 @@ import { BOOKMARKS_MODAL } from '../BookmarksConstants';
 import { translate as __ } from '../../../common/I18n';
 import { noop } from '../../../common/helpers';
 import BookmarkForm from './BookmarkForm';
+import { getBookmarksModalId } from '../../PF4/Bookmarks/BookmarksHelpers';
 
 const SearchModal = ({
+  id,
   setModalClosed,
   onEnter,
   title,
   controller,
   url,
   bookmarks,
+  isPF4,
 }) => (
   <ForemanModal
-    id={BOOKMARKS_MODAL}
+    id={isPF4 ? getBookmarksModalId(id) : BOOKMARKS_MODAL}
```

Comment:
since we count on the `id` in both bookmarks implementations, let's use just the `getBookmarksModalId(id)`
and drop the `isPF4` flag. It will also make it easier to remove the old bookmarks later on

Reply:
Not sure what you mean? `webpack/assets/javascripts/react_app/components/Bookmarks/Bookmarks.js` doesn't pass `id` to the `SearchModal`

Reply:
oh I see, so how about removing the `isPF4` flag and in `getBookmarksModalId` do something like this:
```js
export const getBookmarksModalId = id => id ? `${BOOKMARKS_MODAL}-${id}` : BOOKMARKS_MODAL;
  ```

Reply:
Yeah, changed!